maven-model-builder/src/main/java/org/apache/maven/model/interpolation/AbstractStringBasedModelInterpolator.java

```diff
@@ -133,6 +134,21 @@ public Object getValue(String expression) {
             valueSources.add(new BuildTimestampValueSource(config.getBuildStartTime(), modelProperties));
         }
 
+        valueSources.add(new AbstractValueSource(false) {
+            @Override
+            public Object getValue(String expression) {
+                if ("rootdir".equals(expression)) {
+                    Path path = config.getRootdir();
+                    if (path != null) {
+                        return path.toString();
+                    }
+                    throw new IllegalArgumentException("Unable to find the root directory. "
```

Comment:
ISE rather? The expression isn't invalid but the state within, no?

Reply:
Agreed, I'll fix them all.